src/component.ts

```diff
@@ -23,21 +98,30 @@ export class Functionless extends Component {
     this.tsProject.addDevDeps(...Functionless.devDependencies);
     this.tsProject.addDeps(...Functionless.dependencies);
 
-    // TODO: when optional?
-    if (this.tsProject.tsconfig) {
-      // https://github.com/projen/projen/issues/1482
-      // TODO: type
-      const compilerOptions = this.tsProject.tsconfig.compilerOptions as any;
-      const plugins: any[] = compilerOptions.plugins ?? [];
-
-      plugins.push({
-        transform: `${Functionless.coreDependency}/lib/compile`,
-      });
-      plugins.push({
-        name: Functionless.languageServiceDependency,
-      });
-
-      compilerOptions.plugins = plugins;
+    configureTsConfig(this.tsProject.tsconfig);
+    configureTsConfig(this.tsProject.tsconfigDev);
+
+    function configureTsConfig(
+      config?: TypescriptConfig & {
+        compilerOptions: TypeScriptCompilerOptionsExtended;
+      }
+    ) {
+      if (config) {
+        const plugins =
+          // @ts-ignore - hack: write to readonly field (JSII forces us to use readonly)
+          (config.compilerOptions.plugins = config.compilerOptions.plugins
+            ? // copy the array because tsconfig and tsconfigDev are sharing the same data
+              // we don't want to rely on that always being true
+              [...config.compilerOptions.plugins]
+            : []);
+
+        plugins.push({
+          transform: `${Functionless.coreDependency}/lib/compile`,
+        });
+        plugins.push({
+          name: Functionless.languageServiceDependency,
+        });
```

Comment:
Do I need to configure the language service plugin on tsconfig.dev.json, problably not?

Reply:
Dev is used for the tests. Don't think it'll be ready by the ide unless they change it